drivers/usb/udc/udc_dwc2_vendor_quirks.h

```diff
@@ -607,24 +607,29 @@ static inline int esp32_usb_otg_init(const struct device *dev,
 	return ret;
 }
 
-static inline int esp32_usb_otg_enable_phy(struct phy_context_t *phy_ctx, bool enable)
+static inline int esp32_usb_otg_enable_clk(struct phy_context_t *phy_ctx)
 {
-	LOG_MODULE_DECLARE(udc_dwc2, CONFIG_UDC_DRIVER_LOG_LEVEL);
+	usb_wrap_ll_enable_bus_clock(true);
 
-	if (enable) {
-		usb_wrap_ll_enable_bus_clock(true);
-		usb_wrap_hal_init(&phy_ctx->wrap_hal);
+	usb_wrap_ll_reset_register();
+	usb_wrap_hal_init(&phy_ctx->wrap_hal);
 
 #if USB_WRAP_LL_EXT_PHY_SUPPORTED
-		usb_wrap_hal_phy_set_external(&phy_ctx->wrap_hal,
-					      (phy_ctx->target == USB_PHY_TARGET_EXT));
+	usb_wrap_hal_phy_set_external(&phy_ctx->wrap_hal, (phy_ctx->target == USB_PHY_TARGET_EXT));
 #endif
 
+	return 0;
+}
+
+static inline int esp32_usb_otg_enable_phy(struct phy_context_t *phy_ctx, bool enable)
+{
+	LOG_MODULE_DECLARE(udc_dwc2, CONFIG_UDC_DRIVER_LOG_LEVEL);
+
+	if (enable) {
+		usb_wrap_ll_phy_enable_pad(phy_ctx->wrap_hal.dev, true);
 		LOG_DBG("PHY enabled");
 	} else {
-		usb_wrap_ll_enable_bus_clock(false);
```

Comment:
Is it intended to never turn off the clock? Maybe you want to use shutdown quirk to disable it.